src/config.c

```diff
@@ -229,17 +229,87 @@ CONFIG_GETTER(getMinOperationWorkers) {
   return sdscatprintf(ss, "%lu", config->minOperationWorkers);
 }
 
-// MT_MODE, WORKER_THREADS
-CONFIG_SETTER(trySetDeprecatedWorkersConfig) {
+/************************************ DEPRECATION CANDIDATES *************************************/
+
+enum MTMode {
+  MT_MODE_OFF,
+  MT_MODE_ONLY_ON_OPERATIONS,
+  MT_MODE_FULL,
+};
+
+// Old configuration
+enum MTMode mt_mode_config = MT_MODE_OFF;
+size_t numWorkerThreads_config = 0;
+
+// WORKER_THREADS
+CONFIG_SETTER(setDeprWorkThreads) {
   RedisModule_Log(RSDummyContext, "warning", "MT_MODE and WORKER_THREADS are deprecated, use WORKERS and MIN_OPERATION_WORKERS instead");
-  int acrc = AC_Advance(ac); // Consume the value argument
-  RETURN_STATUS(acrc);
+  size_t newNumThreads;
+  int acrc = AC_GetSize(ac, &newNumThreads, AC_F_GE0);
+  CHECK_RETURN_PARSE_ERROR(acrc);
+  if (newNumThreads > MAX_WORKER_THREADS) {
+    QueryError_SetErrorFmt(status, QUERY_ELIMIT, "Number of worker threads cannot exceed %d", MAX_WORKER_THREADS);
+    return REDISMODULE_ERR;
+  }
+  numWorkerThreads_config = newNumThreads;
+  return REDISMODULE_OK;
 }
 
-CONFIG_GETTER(getDeprecatedWorkersConfig) {
-  return sdsnew("Deprecated, see WORKERS and MIN_OPERATION_WORKERS");
+CONFIG_GETTER(getDeprWorkThreads) {
+  RedisModule_Log(RSDummyContext, "warning", "MT_MODE and WORKER_THREADS are deprecated, use WORKERS and MIN_OPERATION_WORKERS instead");
+  sds ss = sdsempty();
+  size_t numThreads;
+  switch (mt_mode_config) {
+  case MT_MODE_OFF:
+    numThreads = 0;
+    break;
+  case MT_MODE_ONLY_ON_OPERATIONS:
+    numThreads = config->minOperationWorkers;
```

Comment:
Shouldn't it be `max(config->minOperationWorkers, config->numWorkerThreads)` ?

Reply:
I don’t see why.
As I see it, the old config was controlled by the MT_MODE, so it make sense to return one of the new values depending on the requested mode

Reply:
What if the default `minOperationWorkers` is 4, and the user set `MT_MODE_ONLY_ON_OPERATIONS` and `WORKER_THREADS 2` ? In reality, he would have 4 threads running no?

Reply:
No, because in the upgrade phase we set the min operation workers configuration to 2, and keep the workers 0